flutter/pubspec.yaml

```diff
@@ -15,6 +15,7 @@ dependencies:
     sdk: flutter
   sentry: ^5.0.0
   package_info_plus: ^1.0.0
+  device_info_plus: ^1.0.0
```

Comment:
If we don't want to depend on this we could consider making this available as a seperate package.

Reply:
We try to keep SDKs dependency-free as much as possible but here I don't see a way around it. Since usually this is data we get from the platform itself, and in Flutter (correct me if I'm wrong please) `device_info_plus` became the facto platform information library. Folks from [Flutter Community maintaining this library](https://github.com/fluttercommunity/plus_plugins) as very responsive and approachable so landing a patch there it's surely easier than trying to land a patch some library that's unmaintained or not so mainstream.

@marandaneto what do you think?
Also, for visibility: @mitsuhiko

Reply:
there are 2 ways to go and both have their trade-offs.
we're not depending only on `device_info_plus` if we want to go fully native with Web/Desktop support, we'll depend on a bunch of them that might conflict with App's dependencies, that's one of them.

Indeed, the `plus` plugins are well maintained and welcoming new contributors, and accepting/merging/releasing PRs quite fast.

Going this way, we'd be missing features that the Native SDKs offer ootb, eg, Offline caching, Release health, etc, also maintaining another code base.

Going with the Native SDKs, like adding sentry-native and sentry-javascript for JS and Desktop support seems less code and easier to do, but obviously, it has its trade-offs as well.

Let's discuss this before going forward.

Reply:
Until we decided wether we want to use `device_info_plus` or write the native code ourself, we could also just include information given by Dart and Flutter.

So the question is, is this `Enricher` approach something which we want to move on with?

Reply:
I think it would be great to have something like this available. Maybe it should be in a separate extension package so developers are not forced to use it.

Reply:
@kuhnroyal You can already use event processors to add information to your crash reports and events. See https://pub.dev/documentation/sentry/latest/sentry/EventProcessor.html